lib/dao.js

```diff
@@ -380,7 +380,15 @@ DataAccessObject.create = function(data, options, cb) {
       obj.trigger('save', function(saveDone) {
         const _idName = idName(Model);
         let val = Model._sanitizeData(obj.toObject(true), options);
-        function createCallback(err, id, rev) {
+
+        // SPIKE: changed the signature
+        function createCallback(options) {
+
+          const { err, id, rev, propertyCoercer } = options;
+
+          // SPIKE: apply coercion
+          propertyCoercer(obj.__data, Model.definition.properties);
```

Comment:
I find this approach as too complex and error-prone. We have to remember to call coercion in every DAO method that's dealing with model data coming from the connector.

Have you considered applying the coercion rules inside [ModelBaseClass.prototype._initProperties](https://github.com/strongloop/loopback-datasource-juggler/blob/261dd1c865e8b197882a161b909b90b0e6c3c2a3/lib/model.js#L202-L207), as I explained in https://github.com/strongloop/loopback-connector-mongodb/pull/591#discussion_r473908112?

Reply:
@bajtos I got your concern. 
The `toDatabase` coercion is already handled in mongodb connector(for both decimal128 and objectid), but for the other direction `fromDatabase`, what's the current situation? 
If currently an objectid property passed from mongodb will be automatically converted to string in juggler, and it's a common logic for any custom type from any connectors, then I agree we should keep juggler untouched.

Reply:
MongoDB connector already implements a similar functionality for `decimal128` type. It uses the following property definition:

```js
{ 
  type: 'string', 
  mongodb: {dataType: 'decimal128'},
}
```

The connector takes care of converting string values coming from juggler to MongoDB's `Decimal128` instances, and converting `Decimal128` instances coming from MongoDB back to string before they are passed to juggler.